### PR TITLE
Update/data cy

### PR DESCRIPTION
**Pull request by yoslie**
## Overview
Add and update data cy attribute

### Changes
- [set specific data-cy on modal choose template](https://github.com/jabardigitalservice/j-site-builder/commit/376239203f88ca53cd5a6502d1ba4c34807c07be)
- [add data cy in selector button setup widget](https://github.com/jabardigitalservice/j-site-builder/commit/32efbb903f48beb4fc37889960789bc342ef3994)
- [add data cy in component no data](https://github.com/jabardigitalservice/j-site-builder/commit/88f2c6228aac5f6a30d89602b556a53a6df86b21)
- [add data cy to modal component](https://github.com/jabardigitalservice/j-site-builder/commit/35f544dea6e5c3758ecad8029e0b4c605c319beb)
- [add data cy to widget slideshow component](https://github.com/jabardigitalservice/j-site-builder/commit/ea767ed79594c6e6bffe3e754be7288b28ed76aa)
- [add data cy to widget agenda component](https://github.com/jabardigitalservice/j-site-builder/commit/0cf8fbb79d50a545be8967e86be56068220d6b45)
- [add data cy to widget container section](https://github.com/jabardigitalservice/j-site-builder/commit/5b266f8f4c889a22916ae175b9247dfb803414d6)
- [add data cy to modal input title section](https://github.com/jabardigitalservice/j-site-builder/commit/0d9aa74c01398fcb7d723a4d1676b17c2b58d7bb)
- [add data cy in widget setup gallery](https://github.com/jabardigitalservice/j-site-builder/commit/ab84a55dae7309ea54dd27ca9d70b5c7cb84f7b5)

### Related
- [Q1A1 - Membuat test script feature slideshow](https://app.clickup.com/t/9003239225/CES-3220)
- [Q1A4 - Membuat test script feature widget gallery](https://app.clickup.com/t/9003239225/CES-3688)
- [Q1A5 - Membuat test script feature widget input judul](https://app.clickup.com/t/9003239225/CES-3693)
- [Q1A6 - Membuat test script feature widget agenda](https://app.clickup.com/t/9003239225/CES-4186)

## Evidence
title: update data cy in modal select template and add data cy in base component (modal, nodata), and widget (gallery, agenda, slideshow, modal section title)
project: J-Site
participants: @yoslie @Ibwedagama @marsellavaleria19 @doohanas

### CODE REVIEW

File: components/Page/BuilderWidgetContainer.vue [hunk 1]
```diff
@@ -21,10 +22,20 @@
         >
           <span
             class="line-clamp-1 max-w-[150px] self-center font-lato text-[12px] font-medium leading-[23px] text-gray-900"
+            :data-cy="`j-site-page-builder__title-section-${
+              sectionTitle
+                ? sectionTitle.toLowerCase().replaceAll(' ', '-')
```
Comment (Ibwedagama) [inline components/Page/BuilderWidgetContainer.vue:27]:
```suggestion
                ? sectionTitle?.toLowerCase().replaceAll(' ', '-')
```

File: components/Page/BuilderWidgetSelector.vue [hunk 2]
```diff
@@ -33,7 +33,12 @@
       Setup element sesuai dengan kebutuhan kamu
     </p>
     <div class="max-w-[200px]">
-      <UButton @click="toggleConfig">
+      <UButton
+        :data-cy="`j-site-page-builder__button-widget-setup-${props.widgetName
+          .toLowerCase()
+          .replaceAll(' ', '-')}`"
```
Comment (Ibwedagama) [inline components/Page/BuilderWidgetSelector.vue:39]:
```suggestion
        :data-cy="`j-site-page-builder__button-widget-setup-${props.widgetName?
          .toLowerCase()
          .replaceAll(' ', '-')}`"
```

File: components/Page/ModalChooseTemplate.vue [hunk 3]
```diff
@@ -61,15 +61,19 @@
                     :src="template.thumbnail"
                     :alt="template.name"
                     class="h-full w-full object-cover"
-                    data-cy="j-site-page-template__image-thumbnail"
+                    :data-cy="`j-site-page-template__image-thumbnail-${template.name
+                      .toLowerCase()
+                      .replaceAll(' ', '-')}`"
```
Comment (Ibwedagama) [inline components/Page/ModalChooseTemplate.vue:66]:
```suggestion
                    :data-cy="`j-site-page-template__image-thumbnail-${template.name?
                      .toLowerCase()
                      .replaceAll(' ', '-')}`"
```

**Comment by Ibwedagama** [comment]
hello @yoslie please add conditional chaining `?` before running `toLowerCase()` because if the value is null or undefined it will crash the application

**Review by marsellavaleria19 — APPROVED** [review]

**Comment by yoslie** [comment]
> hello @yoslie please add conditional chaining `?` before running `toLowerCase()` because if the value is null or undefined it will crash the application

hi, thanks @Ibwedagama I already add it 🙏

**Review by Ibwedagama — APPROVED** [review]

**Review by doohanas — APPROVED** [review]

**Pull request merged**